src/components/sidebar.js

```diff
@@ -2,36 +2,54 @@ import React, {useEffect} from 'react';
 import Chart from 'chart.js';
 
 import './sidebar.scss';
-import Red from '../icons/attrs/red.png';
-import Yellow from '../icons/attrs/yellow.png';
-import Blue from '../icons/attrs/blue.png';
+import OffensiveMajorIcon from '../icons/attrs/offensiveMajor.png';
+import OffensiveMinorIcon from '../icons/attrs/offensiveMinor.png';
+import DefensiveMajorIcon from '../icons/attrs/defensiveMajor.png';
+import DefensiveMinorIcon from '../icons/attrs/defensiveMinor.png';
+import UtilityMajorIcon from '../icons/attrs/utilityMajor.png';
+import UtilityMinorIcon from '../icons/attrs/utilityMinor.png';
 import {Col, Row} from 'react-bootstrap';
 
-export default function Sidebar() {
+const red = '#ED4040';
+const blue = '#70B0F2';
+const yellow = '#E9C16B';
+const transparent = '#00000000';
+
+export default function Sidebar(props) {
+    let filledSkillCells = [];
+    for(let i = 0; i < props.majorAttributes.utility; i++) {
+        filledSkillCells.push(<Col className="skillCell filled"/>);
+    }
+
+    let emptySkillCells = [];
+    for(let i = props.majorAttributes.utility; i < 6; i++) {
+        emptySkillCells.push(<Col className="skillCell"/>);
+    }
+
```

Comment:
I think this could be simplifies to one loop. 
`
let skillCells = [];
for (let i = 0; i < 6; i++) {
    if (i < props.majorAttributes.utilitiy) {
        skillCells.push(<Col className="skillCell filled"/>);
    }
    else {
        skillCells.push(<Col className=skillCell" />);
    }
}`

If you adopt this you will need to update the place you are rendering them to use the new single list.